lms/djangoapps/program_enrollments/api/reading.py

```diff
@@ -412,15 +415,19 @@ def get_users_by_external_keys_and_org_key(external_user_keys, org_key):
             saml_provider.get_social_auth_uid(external_user_key)
             for external_user_key in external_user_keys
         }
-        social_auths = UserSocialAuth.objects.filter(uid__in=social_auth_uids)
-        found_users_by_external_keys.update({
-            saml_provider.get_remote_id_from_social_auth(social_auth): social_auth.user
-            for social_auth in social_auths
-        })
+        if social_auth_uids:
+            # Filter should be case insensitive
+            query_filter = reduce(or_, [Q(uid__iexact=uid) for uid in social_auth_uids])
+            social_auths = UserSocialAuth.objects.filter(query_filter)
+            found_users_by_external_keys.update({
+                saml_provider.get_remote_id_from_social_auth(social_auth).lower(): social_auth.user
+                for social_auth in social_auths
+            })
```

Comment:
I think this might be a huge performance hit

UserSocialAuth is based on AbstractUserSocialAuth
uid on that is an indexed field so the original query would be pretty performant, it's just checking the index
the new query probably has to do a tablescan

Reply:
Played around with this a bit - for MySQL, this is equivalent to `SELECT * FROM usersocialauth WHERE uid LIKE external_user_key`, so it should still be able to take advantage of the index since there are no wildcard characters. Tested it with the read-only DB and it didn't seem to affect performance.

Reply:
looks like mysql only has LIKE not ILIKE and mysql is often set up by default so that the column's LIKE is case insensitive, that seems to be the case here